E0259 needs to be updated to new format

From: src/test/compile-fail/E0259.rs

E0259 needs a span_label, updating it from:

```
error[E0259]: an extern crate named `collections` has already been imported in this module
  --> src/test/compile-fail/E0259.rs:12:1
   |
11 | extern crate collections;
   | ------------------------- previous import of `collections` here
12 | extern crate libc as collections; //~ ERROR E0259
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0259]: an extern crate named `collections` has already been imported in this module
  --> src/test/compile-fail/E0259.rs:12:1
   |
11 | extern crate collections;
   | ------------------------- previous import of `collections` here
12 | extern crate libc as collections; //~ ERROR E0259
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ already imported
```

--- a/src/librustc_resolve/lib.rs
+++ b/src/librustc_resolve/lib.rs
@@ -3366,7 +3366,11 @@ impl<'a> Resolver<'a> {
         };
 
         let mut err = match (old_binding.is_extern_crate(), binding.is_extern_crate()) {
-            (true, true) => struct_span_err!(self.session, span, E0259, "{}", msg),
+            (true, true) => {
+                let mut e = struct_span_err!(self.session, span, E0259, "{}", msg);
+                e.span_label(span, &format!("`{}` was already imported", name));
+                e
+            },
             (true, _) | (_, true) if binding.is_import() || old_binding.is_import() => {
                 let mut e = struct_span_err!(self.session, span, E0254, "{}", msg);
                 e.span_label(span, &"already imported");

--- a/src/test/compile-fail/E0259.rs
+++ b/src/test/compile-fail/E0259.rs
@@ -9,6 +9,10 @@
 // except according to those terms.
 
 extern crate collections;
-extern crate libc as collections; //~ ERROR E0259
+//~^ NOTE previous import of `collections` here
+
+extern crate libc as collections;
+//~^ ERROR E0259
+//~| NOTE `collections` was already imported
 
 fn main() {}

